src/boxed.rs

```diff
@@ -647,39 +647,25 @@ impl<'a, F: ?Sized + Future + Unpin> Future for Box<'a, F> {
     }
 }
 
-macro_rules! array_impls {
-    ($($N: expr)+) => {
-        $(
-            /// This impl replaces unsize coercion.
-            impl<'a, T> From<Box<'a, [T; $N]>> for Box<'a, [T]> {
-                fn from(mut arr: Box<'a, [T; $N]>) -> Box<'a, [T]> {
-                    let ptr = core::ptr::slice_from_raw_parts_mut(arr.as_mut_ptr(), $N);
-                    mem::forget(arr);
-                    unsafe { Box::from_raw(ptr) }
-                }
-            }
-
-
-            /// This impl replaces unsize coercion.
-            impl<'a, T> TryFrom<Box<'a, [T]>> for Box<'a, [T; $N]> {
-                type Error = Box<'a, [T]>;
-                fn try_from(mut slice: Box<'a, [T]>) -> Result<Box<'a, [T; $N]>, Box<'a, [T]>> {
-                    if slice.len() == $N {
-                        let ptr = slice.as_mut_ptr() as *mut [T; $N];
-                        mem::forget(slice);
-                        Ok(unsafe { Box::from_raw(ptr) })
-                    } else {
-                        Err(slice)
-                    }
-                }
-            }
-        )+
+/// This impl replaces unsize coercion.
```

Comment:
In case you aren't aware: there is the experimental [`CoerceUnsized`](https://doc.rust-lang.org/core/ops/trait.CoerceUnsized.html) trait for exactly this, but I don't know if it's desirable to use it for this crate.

Reply:
I'm aware, thanks, but I don't want to introduce nightly-only APIs/features to this crate.